File: spring-boot-jooq/src/main/java/com/java/springbootjooq/SampleService.java
package com.java.springbootjooq;

import com.tej.JooQDemo.jooq.sample.model.Tables;
import org.jooq.DSLContext;
import org.springframework.stereotype.Service;

import com.tej.JooQDemo.jooq.sample.model.tables.pojos.Book;

import java.util.List;

@Service
public class SampleService {

    private DSLContext dslContext;

    public SampleService(DSLContext dslContext) {
        this.dslContext = dslContext;
    }

    public void insertBook(Book book) {
        dslContext.insertInto(Tables.BOOK, Tables.BOOK.ID, Tables.BOOK.TITLE, Tables.BOOK.AUTHOR)
                .values(book.getId(), book.getTitle(), book.getAuthor())
                .execute();
    }

    public List<Book> getBooks() {
        return dslContext.selectFrom(Tables.BOOK)
                .fetchInto(Book.class);
    }

}

File: kotlin-jook/src/main/kotlin/com/test/kotlinjooq/SampleService.kt
package com.test.kotlinjooq

import com.test.kotlinjooq.db.mysql.tables.pojos.Book
import org.springframework.stereotype.Service

import com.test.kotlinjooq.db.mysql.tables.references.*
import org.jooq.DSLContext
import org.springframework.beans.factory.annotation.Autowired

@Service
class SampleService(
    @Autowired private val dslContext: DSLContext
) {

    fun insertBook(book: Book) {
        dslContext.insertInto(BOOK, BOOK.ID, BOOK.TITLE, BOOK.AUTHOR)
            .values(book.id, book.title, book.author)
            .execute()
    }

    fun getBooks() : List<Book> {
        return dslContext.selectFrom(BOOK)
            .fetchInto(Book::class.java)
    }

}